File: algorithms/1_easy/C/03274_check_if_two_chessboard_squares_have_the_same_color/solution.c
bool checkTwoChessboards(const char *c1, const char *c2) {
    return (((c1[0] - '0') + (c1[1] - 'a')) & 1) == (((c2[0] - '0') + (c2[1] - 'a')) & 1);
}

File: algorithms/1_easy/C/03274_check_if_two_chessboard_squares_have_the_same_color/solution.rs
impl Solution {
    #[inline(always)]
    fn calc_color(b: &[u8])-> u8 {
        ((b[0] - b'0') + (b[1] - b'a')) & 1
    }

    pub fn check_two_chessboards(coordinate1: String, coordinate2: String) -> bool {
        Self::calc_color(coordinate1.as_bytes()) == Self::calc_color(coordinate2.as_bytes()) 
    }
}
